src/BenchmarkDotNet/Reports/Metric.cs

```diff
@@ -6,11 +6,11 @@ namespace BenchmarkDotNet.Reports
 {
     public class Metric
     {
-        public double Value { get; }
+        public double? Value { get; }
 
         public IMetricDescriptor Descriptor { get; }
 
-        public Metric(IMetricDescriptor descriptor, double value)
+        public Metric(IMetricDescriptor descriptor, double? value)
```

Comment:
This is a breaking change in the public API. I thought about using `NaN` instead of nullable, but I'm not sure if that is already a valid metric value.

Reply:
Or another option is to revert the `Metric` changes, and just don't add the metric if the value couldn't be calculated (instead of adding null), but that may omit the column entirely if none of the benchmarks can calculate it, and I don't think that's desirable (probably better to see a column of all `?` rather than no column at all).

Reply:
I opted for the latter option. I reverted `Metric.cs` to not break the public API, don't add the metric if the allocation could not be calculated, and force the allocated column to show if the memory diagnoser is attached (in case no allocations were measured for any benchmark).